models/authentication/agent/agent-connected/index.tsx

```diff
@@ -108,8 +113,37 @@ export class AgentConnected {
     const superAgentsListScopesRaw = await superAgentsList.getScopeForAgent(
       this.email
     );
+    s3Scopes = [...superAgentsListScopesRaw];
     superAgentScopes.add(superAgentsListScopesRaw);
 
+    // Compare scopes from D-Roles and S3, log differences to Sentry
+    if (process.env.D_ROLES_ENABLED === 'enabled' && drolesScopes.length > 0) {
+      const drolesScopesSet = new Set(drolesScopes);
+      const s3ScopesSet = new Set(s3Scopes);
+
+      const onlyInDroles = drolesScopes.filter(
+        (scope) => !s3ScopesSet.has(scope)
+      );
+      const onlyInS3 = s3Scopes.filter((scope) => !drolesScopesSet.has(scope));
```

Comment:
Simpler way to compare them could be https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/union

Union both set and compare lenght of union with length of any of the two sets

Reply:
But you would have the `onlyInDroles` and `onlyInS3` variables